_transcripts/028-rust-1.44-1.45.md

```diff
@@ -0,0 +1,794 @@
+support nowadays is in fact mature enough to support this. It's just, the Rust
+library team doesn't want to actually expose this yet, until they're absolutely
+sure that const is ready for prime time.
+
+__Jon__: Yeah, or at least some subset of const that were—
+
+__Ben__: Yeah, you mentioned before— you want to go over, like, the arbitrary
+restrictions that we have right now, to keep const from being too exposed?
+
+__Jon__: Yeah, it's kind of silly, but also kind of cool. So the idea is that
+internally, the compiler is allowed to use unstable features because it doesn't
+actually leak out into the application code. This is what allows the compiler
+to, like, implement new features, and then start testing them out internally,
+even on stable, because the compiler is compiled with nightly. And this means
+that internally in the compiler, you have access to const generics, so you can
+easily write, like, an implementation of `From<[T;___]>`, like an array of `T`s
+of any length. And for `Vec<T>`. You can write that implementation for any N,
+using const generic in the compiler. But if you did, you're sort of leaking the
+fact to the world that you're using this feature, and that this feature exists.
+And there are a couple of reasons why you don't want to do this. But the way
+they've come up with not leaking this value is that there's a trait in the
+standard library called `LengthAtMost32`, which is only implemented for arrays
+of length that is less than or equal to 32. And so what they have is they have
+this impl of `From<[T; N] for Vec<T>`. But only where the array of that length
+implements that trait. And the only thing this trait does is limit what
+implementations that applies to. And so the moment that const generics are
+actually made stable, they could just remove that trait from the impl, and
+suddenly the implementation is now visible to the whole world for any N rather
+than just Ns up to 32.
+
+__Ben__: Yeah, and so, as kind of a precursor to lifting that restriction,
+getting back to my point, and the radical proposal I have, you can now call
+`Vec::from` with just an array, using an array literal, a stack allocated array.
+And I think this is a great sort of mirror to `String::from`. And in fact, if it
+would finally become stable for, you know, arrays of larger than 32, I suggest
+that maybe it should replace the `vec!` macro, which again, highly radical and
+contentious—
+
+__Jon__: Very radical of you.
+
+__Ben__: Only because I think macros are a little bit, kind of, opaque to
+editors, and they kind of don't play super nicely with incremental compilation.
+And to be clear, none of these are really deal breakers for using the `vec!`
+macro specifically, which is a pretty simple macro. But when has something ever
+being not a big deal stopped programmers from arguing about something.
+
+__Jon__: That's right. Wait. So isn't then the galaxy brain solution here, to
```

Comment:
"galaxy brain"?  Did I hear that right?

Reply:
[That's right!](https://knowyourmeme.com/memes/galaxy-brain)